ui/app/routes/vault/cluster/oidc-provider.js

```diff
@@ -94,6 +94,7 @@ export default class VaultClusterOidcProviderRoute extends Route {
   _handleSuccess(response, baseUrl, state) {
     const { code } = response;
     const redirectUrl = this._buildUrl(baseUrl, { code, state });
+    this.redirect_uri = redirectUrl;
```

Comment:
Instead of setting this, we can try to replace the location and return the `redirectUrl` at the end if it is blocked by the browser. So we would change both this and the `_handleError` block below to:
```
if (!Ember.testing) {
  this.win.location.replace(redirectUrl);
}
return { redirectUrl };
```